### PR TITLE
Updated test-infra-trusted build cluster access

### PR DESCRIPTION
**Pull request by chaodaiG**
/hold
Will need to delete test-infra-trusted from merged kubeconfig first

/cc @fejta @cjwagner

### CODE REVIEW

File: config/prow/cluster/pipeline_deployment.yaml [hunk 1]
```diff
@@ -25,7 +25,7 @@ spec:
         env:
         # Use KUBECONFIG envvar rather than --kubeconfig flag in order to provide multiple configs to merge.
         - name: KUBECONFIG
-          value: "/etc/kubeconfig/config"
+          value: "/etc/kubeconfig/config:/etc/kubeconfig-build-test-infra-trusted/kubeconfig"
```
Comment (cjwagner) [inline config/prow/cluster/pipeline_deployment.yaml:28]: Volume mounts are missing in this file.

**Comment by chaodaiG** [comment]
/unhold
Done

File: config/prow/cluster/pipeline_deployment.yaml (hunk 1 above)
Comment (chaodaiG) [reply config/prow/cluster/pipeline_deployment.yaml]: This file is not deployed as far as I can tell, reverted changes in this file to avoid confusion

**Review by cjwagner — APPROVED** [review]

**Comment by chaodaiG** [comment]
The postsubmit job for deploying prow runs on `test-infra-trusted` cluster, so it will not succeed, I will manually apply changes to these files once this PR is merged

**Pull request merged**